### ISSUE TITLE
linter: (import/no-named-as-default) false positive for default class export from node_modules when class name matches export identifier

### ISSUE THREAD

**Issue by dotSality** [oxc-project/oxc#22319]
### What version of Oxlint are you using?

1.63.0

### What command did you run?

oxlint ./src

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["import"],
  "rules": {
    "import/no-named-as-default": "error",
    "import/no-named-as-default-member": "warn"
  }
}
```

### What happened?

`import/no-named-as-default` incorrectly flags imports of a default-exported class from `node_modules` when the import identifier matches the class name, even though there is no named export with that identifier.

**Case 1 — value import:**
```
import Hls, { CMCDController } from 'hls.js'; // oxc: Module "hls.js" has named export "Hls"
```

**Case 2 — type import:**
```
import type Hls from 'hls.js'; // oxc: Module "hls.js" has named export "Hls"
```

**Source of `hls.js`:**

```
export default class Hls implements HlsEventEmitter { ... }
```
`Hls` is only a default export — there is no `export { Hls }` or `export class Hls`. The rule incorrectly treats the class name as a named export.

**Does not trigger with a local file:**
```
// mock-class.ts
export default class MockClass {
  constructor() {}
}

// consumer.ts
import MockClass from "./mock-class"; // no error
```
Probably the bug is specific to module resolution for node_modules, not local files

**Expected**: No error. ESLint with the same config does not flag these cases.

Might have been related to [#19501](https://github.com/oxc-project/oxc/issues/19501) which is closed, but did not cover this specific case — class name used as default export identifier with no actual named export, and `import type`
